### PR TITLE
fix(Popup) popup appearing in wrong position when open prop is set 

### PR DESCRIPTION
**Pull request by cdaringe**
# problem

- please see #2389

# solution

- add in RAF behavior per @levithomason's recommendation.
  - i _didn't_ actually properly animate it in, but this still has the added benefit of keeping the `<Popup />` correctly attached to the trigger if the trigger moves

## demo

here i use `<Popup />` for form validation.

with fix, the `<Popup />` immediately renders when the field is dirty and empty:

good mov

without fix.  the `<Popup />` is out of the viewport.  I have to fidget with the `trigger={<Input />}` to get the Popup to eventually appear where it has been commanded to go:

bad mov

### CODE REVIEW

File: src/modules/Popup/Popup.js [hunk 1]
```diff
@@ -341,6 +344,21 @@ export default class Popup extends Component {
...
+      this.triggerRef = triggerRef
+    }
+  }
+
+  setPosition = () => {
+    if (this.triggerRef) {
+      this.setPopupStyle(this.props.position)
+    }
+
+    this.animationRequestId = requestAnimationFrame(this.setPosition)
```
Comment (fracmak) [inline src/modules/Popup/Popup.js:360]: haven't you introduced an infinite loop now? When does this ever get short circuited?

Comment (cdaringe) [reply src/modules/Popup/Popup.js:360]: good point.  fixed in 7c2104f

**Comment by cdaringe** [comment]
hi all, kind and cordial bump!

**Comment by levithomason** [comment]
Such a gentleman 😉 reviewing!

**Comment by levithomason** [comment]
Just merged master, we can ship this once tests pass again.

**Comment by cdaringe** [comment]
codecov seems to be all hung up!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.80.1`.

**Comment by cdaringe** [comment]
nice!  ps, if you havent seen already, the latest version of semantic-release auto posts back into PRs what versions PRs get released in. so great!

**Comment by aminland** [comment]
With this commit, scrolling the window causes very strange behavior.

semantic-ui-popup-bug

File: src/modules/Popup/Popup.js (hunk 1 above)
Comment (miroslavstastny) [reply src/modules/Popup/Popup.js:360]: Unfortunately the infinite loop is still there - all the time the popup is displayed `setPosition()` is called in a loop, it then calls `setPopupStyle` which re-sets component's state which causes re-render - takes ~20% CPU when the popup is displayed.

**Comment by prawdziwy-sok** [comment]
Hi, this issue is resolved in v0.80.2, but in the newest (v0.82.1 at the moment) it is still present. Did the fix get removed in the meantime somewhere? Could it be put back?

**Pull request merged**